src/lib.rs

```diff
@@ -25,12 +111,23 @@ mod tests {
 
     #[test]
     fn test_map() {
```

Comment:
I tried creating multiples tests with different parameters, but then breaks OnceLock.

Reply:
I think there are choices here are to shift the test coverage  to a higher level and lower lower level tests. The current Smartmodule interface makes this kind of thing a little difficult to work with.   High could go to coverage w/ `smdk tests` like you added in the Makefile; or make those into integration tests with something like [assert_cmd](https://crates.io/crates/assert_cmd).

Lower coverage would be to refactor some of the functionality into utility functions/code used by the smartmodule impl with unit tests testing that functionality.

Reply:
I created e2e tests using commands direct to the Makefile, what do you think?

Reply:
Those are a decent set of tests for a developer to use.  It might be nice if the outputs could be verified. Even if its just diffing with a known fixed output.   Then we might be able to use it for automated test CI.

Reply:
I am a bit confused here, which “output” are we talking about?
The snapshot test output? The function output?

Should I move everything to another function to test it with unit tests?

Reply:
No, sorry, let me clarify.  For example this tests that it functionally completes, but doesn't check the output. This make targets is ok for developer inspected tests, but capturing the output and checking it would let us catch a change in CI automation.   That could done by diffing the output against a saved verifed output, or something more programmatic logic w/ something like the assert_cmd crate.

```
test-comma-readable: build
	smdk test --file ./test-data/comma/input.csv --raw | jq
```

Reply:
Oh, I see, have you checked the new file `tests/e2e.rs`? I think that it's doing it now.

Reply:
`cargo test` now is comparing the Makefile commands output with the outputs files.